src/SCRIPTS/BF/background.lua

```diff
@@ -1,48 +1,26 @@
-SCRIPT_HOME = "/SCRIPTS/BF"
-
-assert(loadScript(SCRIPT_HOME.."/MSP/common.lua"))()
-
 local INTERVAL          = 50         -- in 1/100th seconds
 local MSP_SET_RTC       = 246
 local MSP_TX_INFO       = 186
-local sensorName        = "Tmp1"     -- T1 is never 0 in Betaflight
 
 local lastRunTS
 local timeIsSet = false
-local sensorId
 local mspMsgQueued = false
 
-local function getTelemetryId(name)
-    local field = getFieldInfo(name)
-    if field then
-      return field['id']
-    else
-      return -1
-    end
-end
-
 local function init()
-    sensorId = getTelemetryId(sensorName)
     lastRunTS = 0
 end
 
 local function run_bg()
     -- run in intervals
     if lastRunTS == 0 or lastRunTS + INTERVAL < getTime() then
-
         mspMsgQueued = false
-
         -- ------------------------------------
         -- SYNC DATE AND TIME
         -- ------------------------------------
-
-        -- get sensor value
-        local newSensorValue = getValue(sensorId)
-
-        if not timeIsSet and type(newSensorValue) == "number" and newSensorValue > 0 then
+        if not timeIsSet and protocol.push() then
```

Comment:
This doesn't work (at least on SmartPort), since the telemetry connection is never actually lost when the flight controller is rebooted but the RX isn't power cycled.